### PR TITLE
33 pagination

### LINKED ISSUE #33: [API i Front] Paginacja

### PR DESCRIPTION
**Pull request by mmiszy**
Fixes #33

### CODE REVIEW

File: pages/[displayStyle]/[page].tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      await getArticlesPaginationForGrid(prisma),
+      await getArticlesPaginationForList(prisma),
+    ]);
+
+    const paths = [
+      ...gridPages.map((page) => ({ params: { displayStyle: 'grid' as const, page } })),
+      ...listPages.map((page) => ({ params: { displayStyle: 'list' as const, page } })),
+    ];
+
+    console.log({ gridPages });
```
Comment (takatejr) [inline pages/[displayStyle]/[page].tsx:41]: Hey

Comment (AdamSiekierski) [inline pages/[displayStyle]/[page].tsx]: 👀

**Review by AdamSiekierski — APPROVED** [review]

**Review by drillprop — APPROVED** [review]

**Comment by mmiszy** [comment]
#### 📝Changed routes:


| [**/\[displayStyle\]/\[page\]**](https://polskifrontend-ft2brqtx3.vercel.app/[displayStyle]/[page]) | [**/\[displayStyle\]**](https://polskifrontend-ft2brqtx3.vercel.app/[displayStyle]) |
|:-:|:-:|
|<a href="https://polskifrontend-ft2brqtx3.vercel.app/[displayStyle]/[page]">Screenshot of /[displayStyle]/[page]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/h05MhLGPqneyHHpCJ7hg3MzegOc_.png">(view full size)</a> |<a href="https://polskifrontend-ft2brqtx3.vercel.app/[displayStyle]">Screenshot of /[displayStyle]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/MTZQ8ehUzsH2DJEyvJg7beEat3s_.png">(view full size)</a> |





1 deleted route:
- **/[displayStyle]**

Commit c76fb1739f0df8814be2187d069e393c607481fa (https://polskifrontend-ft2brqtx3.vercel.app).

File: pages/api/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { withAsync } from '../../api-helpers/api-hofs';
-import { prisma } from '../../api-helpers/db';
 
 export default withAsync((_req) => {
-  return prisma.$queryRaw('SELECT 1 + 1;');
+  return null;
+  // return prisma.$queryRaw('SELECT 1 + 1;');
   // return prisma.$queryRaw('SELECT * FROM "User";');
```
Comment (AdamSiekierski) [inline pages/api/index.ts:6]: Dlaczemu tak?

File: api-helpers/db.ts [hunk 3]
```diff
@@ -1,77 +1,40 @@
...
+  const res = await fetch(`https://api.heroku.com/apps/polskifrontend/config-vars`, {
+    headers: {
+      Accept: 'application/vnd.heroku+json; version=3',
+      Authorization: `Bearer ${process.env.HEROKU_API_KEY!}`,
+    },
+  });
+  return res.json() as Promise<{ readonly DATABASE_URL: string }>;
+};
 
 let mutableOpenConnections = 0;
```
[the remark this answers is not in the document]
Comment (mmiszy) [reply api-helpers/db.ts:13]: W czasie builda i w trakcie używania API.

File: types.ts [hunk 4]
```diff
@@ -20,3 +20,9 @@ export type InferGetStaticPathsResult<T> = T extends (
   : never;
 
 export type InferGetStaticPropsContext<T> = GetStaticPropsContext<InferGetStaticPathsResult<T>>;
+
+export type InferGetStaticPropsType2<T> = T extends (
+  ...args: readonly any[]
+) => Promise<{ readonly props: infer P }>
+  ? NonNullable<P>
+  : never;
```
Comment (wisnie) [inline types.ts:28]: Jaka jest różnica między natywnym `InferGetStaticPropsType`, a tą  konstrukcją?

Comment (mmiszy) [reply types.ts:28]: Natywna:
Screenshot 2020-12-22 at 18 18 46

Ta:
Screenshot 2020-12-22 at 18 18 23

**Review by AdamSiekierski — APPROVED** [review]

**Pull request merged**